localstack/utils/cloudformation/template_deployer.py

```diff
@@ -1525,8 +1512,9 @@ def apply_changes(
         if not contains_changes:
             raise NoStackUpdates("No updates are to be performed.")
 
-        # merge stack outputs
+        # merge stack outputs and conditions
         existing_stack.template["Outputs"].update(new_stack.template.get("Outputs", {}))
+        existing_stack.template["Conditions"].update(new_stack.template.get("Conditions", {}))
```

Comment:
@dominikschubert This is essentially the logic change / fix in this PR (remaining changes are mostly refactoring).